packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -252,13 +252,24 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
           if (imports.length) {
             const s = new MagicString(code)
             for (let index = 0; index < imports.length; index++) {
-              const { s: start, e: end, d: dynamicIndex } = imports[index]
+              // To handle escape sequences in specifier strings, the .n field will be provided where possible.
+              const {
+                n: name,
+                s: start,
+                e: end,
+                d: dynamicIndex
+              } = imports[index]
               // check the chunk being imported
-              const url = code.slice(start, end)
+              let url = name
+              if (!url) {
+                const rawUrl = code.slice(start, end)
+                if (rawUrl[0] === `"` && rawUrl[rawUrl.length - 1] === `"`)
```

Comment:
Is it correct here that only double-quotes are taken into account? Otherwise we could use something like this

https://github.com/prettier/plugin-pug/blob/67ac18b0797b2cb91d2ec711ed9721769dbc0e6d/src/utils/common.ts#L167-L174

Reply:
Let's check this in another PR adding a failing test. This seems unrelated to this fix

Reply:
Single-quote will become double-quotes after esbuild transform. But I haven't found any docs to explain this.